File: leetcode/Palindrome_Number.java
import java.util.*;

public class Solution {
    int digit(int x, int pos) {
        while (pos > 0) {
            x /= 10;
            pos--;
        }
        return x % 10;
    }
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        if (x == 0) return true;
        int n = 10;
        int lim = 1000000000;
        while (x < lim) {
            lim /= 10;
            --n;
        }
        for (int i = 0; i < n - 1 - i; ++i)
            if (digit(x, i) != digit(x, n - 1 - i))
                return false;
        return true;
    }
}

File: leetcode/Palindrome_Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;
        if (x < 10)
            return true;
        if (x % 10 == 0)
            return false;
        int y = 0;
        while (x > 0) {
            y = y * 10 + x % 10;
            x /= 10;
            if (x == y)
                return true;
            if (x / 10 == y)
                return true;
            if (x < y)
                break;
        }
        return false;
    }
};
